### PR TITLE
feat: add prettier and a check for it using github actions

### PR DESCRIPTION
**Pull request by naveen521kk**
Forked from #107 
I added prettier as discussed in #107 and formatted the whole codebase. Also, added a check for pull requests also.

### CODE REVIEW

File: .prettierignore
Comment (kartikcho) [inline .prettierignore:2]: nit: add a newline here

**Review by kartikcho — COMMENTED** [review]
The prettier config and workflow looks good to me but let's hold off on merging this until #92 is in.

**Comment by naveen521kk** [comment]
Did I forget to add prettier to dependencies?

**Comment by kartikcho** [comment]
> Did I forget to add prettier to dependencies?
I believe so, yeah.

This is next in line of all our open PRs @naveen521kk so we should ship it, although implementing this on top of #264 and #260 would be easier for those PRs.

**Comment by naveen521kk** [comment]
i will into I tommorrow.

**Comment by saadpasta** [comment]
@naveen521kk @kartikcho I think we should merge this PR after all the remaining PR's get's merged. 
What do you guys say?

**Comment by naveen521kk** [comment]
Just updated with the latest branch.

**Comment by kartikcho** [comment]
@naveen521kk is this blocked by any PR? If not, can we get Prettier into the codebase because formatting has become a mess right now 😅

**Comment by naveen521kk** [comment]
> is this blocked by any PR?

I don't think so. It's been 5 months since this PR :<

Let me fix the merge conflicts.

**Comment by kartikcho** [comment]
Lol no problem, let's clear up backlog PRs this weekend.

**Comment by naveen521kk** [comment]
@kartikcho Please check I didn't mess up anything because I came to this PR after a very long time :)

File: .prettierrc
Comment (kartikcho) [inline .prettierrc:7]: I don't want to start that debate lol but can we use tabs here?

File: .prettierrc
Comment (kartikcho) [inline .prettierrc:5]: Can we set this to true? This is highly subjective so I'll need @saadpasta view on this too.

**Review by kartikcho — COMMENTED** [review]
Only took a look at the prettier and gh action files. Formatting is subjective so I would like @saadpasta to vote on stuff too before we go ahead.

File: .prettierrc
Comment (naveen521kk) [reply .prettierrc:7]: We discussed that previously and came to this https://github.com/saadpasta/developerFolio/pull/107#discussion_r432846616

File: .prettierrc
Comment (naveen521kk) [reply .prettierrc:5]: I love double quotes, I mostly use it, but I'm open to suggestions.

File: .prettierrc
Comment (kartikcho) [reply .prettierrc:7]: Oh, I didn't remember that. Sure, this is alright.

File: .prettierrc
Comment (kartikcho) [reply .prettierrc:5]: This is okay too I guess, no strong opinions really. Let's hold on this until #318 is in (sorry for another rebase 😅)

Comment (naveen521kk) [reply .prettierrc:5]: @kartikcho I think #318 is closed. Can you merge this?

Comment (kartikcho) [reply .prettierrc:5]: I literally just merged a PR haha sorry

Comment (naveen521kk) [reply .prettierrc:5]: will rebase it after few hours.

Comment (naveen521kk) [reply .prettierrc:5]: I rebased it, possibly do a final check and then merge it.

**Review by kartikcho — APPROVED** [review]
Everything looks good, let's get this in!

**Comment by naveen521kk** [comment]
finally merged after 5 months :)

**Pull request merged**
